R/appshot.R

```diff
@@ -40,14 +42,59 @@ appshot.character <- function(app, file = "webshot.png", ...,
     p <- processx::process$new("R", args = c("--slave", "-e", cmd))
   })
 
+  # Make sure app is killed on exit
   on.exit({
     p$kill()
   })
 
   # Wait for app to start
   Sys.sleep(0.5)
 
+  # Get screenshot
   fileout <- webshot(sprintf("http://127.0.0.1:%d/", port), file = file, ...)
 
   invisible(fileout)
 }
+
+
+#' @rdname appshot
+#' @export
+appshot.shiny.appobj <- function(app, file = "webshot.png", ...,
+                              port = getOption("shiny.port"), envvars = NULL) {
+
+
+  port <- available_port(port)
+
+  args <- list(
+    url = sprintf("http://127.0.0.1:%d/", port),
+    file = file,
+    ...
+  )
+  r_session <- callr::r_bg(
+    function(...) {
+      # Wait for app to start
+      Sys.sleep(0.5)
+      webshot::webshot(...)
+    },
+    args
+  )
+
```

Comment:
Maybe add `on.exit(r_session$kill())` to make sure that it goes away? Although now that I think of it, that won't ensure that phantomjs process will exit -- in fact, it's probably the reverse, since if the child R process receives the hard kill signal, it won't be able to send any signals to the phantomjs process. Anyway, something should happen to ensure the child R process and grandchild phantomjs process exit.